### ISSUE TITLE
Accessing status code of successful request.

### ISSUE THREAD

**Issue by orwell1984** [ariebovenberg/quiz#95]
HTTPError has .request, which in turn has status_code. How do I get that in case of non-404 successful request?

**Comment by ariebovenberg** [comment]
Hi @orwell1984,

For now I haven't included the HTTP response itself in the result. Are you only interested in the status code, or potentially other properties as well (e.g. headers)?

**Comment by orwell1984** [comment]
Ideally both, but at present I really need the code.

**Comment by honzajavorek** [comment]
@orwell1984 what is your use case? why do you need it?

**Comment by orwell1984** [comment]
Polling an endpoint from CLI. Code shows me if there is more data to be shown to user.

**Comment by ariebovenberg** [comment]
@orwell1984 It is reasonable to include the request and response as metadata in a result. The only questionable thing is how to expose this on the root `Query` object. Simply adding a property `request` or `response` could clash with `Query` fields having the same name.

Perhaps the least risky is some sort of `__metadata__` attribute, which contains such information. This could contain other info as well (retries, duration, etc.)

In code:

```python
result = quiz.execute(...)
print(result.__metadata__.request)
print(result.__metadata__.response)
```

**Comment by ariebovenberg** [comment]
Additionally, for consistency this should also be available on raw JSON responses.

**Comment by orwell1984** [comment]
Can we expect this in next version?

**Comment by ariebovenberg** [comment]
@orwell1984 sure, this is not to difficult to implement.

**Comment by ariebovenberg** [comment]
released in version 0.1.5. closed.

see the docs here: https://quiz.readthedocs.io/en/latest/advanced.html#response-metadata
